src/libraries/System.Private.CoreLib/src/System/Numerics/Vector4_1.cs

```diff
@@ -0,0 +1,561 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace System.Numerics
+{
+    public readonly struct Vector4<T> : IEquatable<Vector4<T>>, IFormattable
+        where T : struct
+    {
+        // Fields
+        public T X { get; }
+
+        public T Y { get; }
+
+        public T Z { get; }
+
+        public T W { get; }
+
+        // Constructors
+        public Vector4(T value)
+        {
+            X = value;
+            Y = value;
+            Z = value;
+            W = value;
+        }
+
+        public Vector4(T x, T y, T z, T w)
+        {
+            X = x;
+            Y = y;
+            Z = z;
+            W = w;
+        }
+
+        public Vector4(T[] value) : this(value.AsSpan())
+        {
+        }
+
+        public Vector4(T[] value, int offset) : this(value.AsSpan(offset))
+        {
+        }
+
+        public Vector4(ReadOnlySpan<T> value)
+        {
+            ThrowForUnsupportedVectorBaseType();
+
+            if (value.Length < 4)
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeException(
+                    ExceptionArgument.value,
+                    ExceptionResource.ArgumentOutOfRange_SmallCapacity
+                );
+            }
+
+            X = value[0];
+            Y = value[1];
+            Z = value[2];
+            W = value[3];
+        }
+
+        static Vector4()
```

Comment:
We don't want a static constructor here. Instead we should just construct the value at each callsite.

For the software case this won't necessarily be efficient, but once the hardware acceleration is in, it will end up in better codegen for that scenario (which is the mainline one anyways).

Reply:
Fixed.